### PR TITLE
Added Clone Git Repository Option

### PR DESCRIPTION
**Pull request by Deenu143**
Added Clone Git repository option in main screen for cloning GitHub projects

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -114,6 +114,10 @@ dependencies {
     implementation google.gson
     implementation google.guava
     
+    implementation 'io.reactivex.rxjava2:rxjava:2.0.1'
+    implementation 'io.reactivex.rxjava2:rxandroid:2.0.1'
+    implementation "org.eclipse.jgit:org.eclipse.jgit:3.7.1.201504261725-r"
```
Comment (itsaky) [inline app/build.gradle:119]: Dependencies must be declared in `dependencies.gradle` file.

File: app/src/main/java/com/itsaky/androidide/tasks/git/CloneGitTask.java [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+										activity);
+								materialAlertDialogBuilder.setTitle(R.string.git_clone_failed);
+
+								//if file exists
+								if (e.toString().contains("org.eclipse.jgit.api.errors.JGitInternalException")) {
+									materialAlertDialogBuilder
+											.setMessage(e.toString().substring(e.toString().lastIndexOf(":") + 2));
+								}
+								//if missing url
+								else if (e.toString().contains("java.net.MalformedURLException: no protocol")) {
```
Comment (itsaky) [inline app/src/main/java/com/itsaky/androidide/tasks/git/CloneGitTask.java:127]: Can we not simply use `e instanceof JGitInternalException/MalformedURLException` or `e.getCause() instanceof JGitInternalException/MalformedURLException`?

File: app/src/main/java/com/itsaky/androidide/tasks/git/CloneGitTask.java [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+				});
+				progressDialog = materialAlertDialogBuilder.create();
+				progressDialog.show();
+
+				//createObservable
+				createObservable().subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread())
+						.subscribe(new Observer<String>() {
+							@Override
+							public void onSubscribe(Disposable disposable) {
+								Log.d(TAG, "onSubscribe: " + disposable);
```
Comment (itsaky) [inline app/src/main/java/com/itsaky/androidide/tasks/git/CloneGitTask.java:99]: Usage of `android.util.Log` should be replaced with `com.itsaky.androidide.utils.ILogger`.

File: app/src/main/java/com/itsaky/androidide/tasks/git/CloneGitTask.java [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+
+	public String run(String url) throws IOException {
+
+		try {
+			String urlRepo = textInputEditText.getText().toString();
+			uRL = new URL(urlRepo.toString());
+			String fileName = uRL.getFile();
+
+			//clone repo
+			Git.cloneRepository().setURI(uRL + ".git")
```
Comment (itsaky) [inline app/src/main/java/com/itsaky/androidide/tasks/git/CloneGitTask.java:171]: What if the URL already contains `.git` at the end? For example, `https://github.com/AndroidIDEOfficial/AndroidIDE.git`.

File: resources/src/main/res/layout/git_dialog_layout.xml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+		<com.google.android.material.textfield.TextInputEditText
+			android:id="@+id/textRepo"
+			android:layout_width="fill_parent"
+			android:layout_height="wrap_content"
+			android:ems="10"/>
+
+	</com.google.android.material.textfield.TextInputLayout>
+
+</LinearLayout>
```
Comment (itsaky) [inline resources/src/main/res/layout/git_dialog_layout.xml:26]: This whole layout file is redundant. You could replace usage of this file with `layout_dialog_text_input`.

File: resources/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -495,4 +495,17 @@
   <string name="widget_linear_layout_horz">LinearLayout (H)</string>
   <string name="widget_linear_layout_vert">LinearLayout (V)</string>
   <string name="widget_relative_layout">RelativeLayout</string>
+
+  <!-- GIT -->
+  <string name="cancel">Cancel</string>
+  <string name="okay">Okay</string>
```
Comment (itsaky) [inline resources/src/main/res/values/strings.xml:501]: These strings are not needed as they are already available as `android.R.string.ok` and `android.R.string.cancel`.

File: resources/src/main/res/values/strings.xml [hunk 7]
```diff
@@ -495,4 +495,17 @@
...
+  <string name="hide">Hide</string>
+  <string name="clone">Clone</string>
+  <string name="clone_git_repository">Clone Git Repository</string>
+  <string name="git_clone_failed">Git Clone Failed</string>
+  <string name="git_clone_successful">Git Clone Successful</string>
+  <string name="cloning_repository">Cloning Repository</string>
+  <string name="successfully_cloned">Successfully Cloned</string>
```
Comment (itsaky) [inline resources/src/main/res/values/strings.xml:508]: These strings doesn't need to have title case.

File: resources/src/main/res/drawable/ic_git.xml [hunk 8]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:width="24dp"
+    android:height="24dp"
+    android:viewportHeight="24"
+    android:viewportWidth="24">
+    <path
+        android:fillColor="#ffffff"
```
Comment (itsaky) [inline resources/src/main/res/drawable/ic_git.xml:8]: Use `@color/primaryIconColor` instead of `#ffffff`.

**Review by itsaky — CHANGES_REQUESTED** [review]
The source files must have 2-space indentation. See the [contribution guide](https://github.com/AndroidIDEOfficial/AndroidIDE/blob/main/CONTRIBUTING.md).

File: app/src/main/java/com/itsaky/androidide/tasks/git/CloneGitTask.java (hunk 2 above)
Comment (Deenu143) [reply app/src/main/java/com/itsaky/androidide/tasks/git/CloneGitTask.java:121]: I tried but i didn't get the correct information can you fix it

File: app/src/main/java/com/itsaky/androidide/tasks/git/CloneGitTask.java (hunk 3 above)
Comment (Deenu143) [reply app/src/main/java/com/itsaky/androidide/tasks/git/CloneGitTask.java]: I removed it it says cannot be referenced from a static context

File: resources/src/main/res/values/strings.xml (hunk 6 above)
Comment (Deenu143) [reply resources/src/main/res/values/strings.xml]: Okay

File: resources/src/main/res/drawable/ic_git.xml (hunk 8 above)
Comment (Deenu143) [reply resources/src/main/res/drawable/ic_git.xml]: Done

File: resources/src/main/res/values/strings.xml (hunk 7 above)
Comment (Deenu143) [reply resources/src/main/res/values/strings.xml]: Done

File: app/src/main/java/com/itsaky/androidide/tasks/git/CloneGitTask.java (hunk 2 above)
Comment (itsaky) [reply app/src/main/java/com/itsaky/androidide/tasks/git/CloneGitTask.java:120]: Okay

**Pull request merged**